packages/babel-generator/src/printer.ts

```diff
@@ -327,6 +331,12 @@ class Printer {
     this._buf._sourcePosition.identifierName = identifierName;
   }
 
+  getIdentifierName(pos: Pos): string | null {
+    if (this._inputMap && pos) {
+      return originalPositionFor(this._inputMap, pos).name;
```

Comment:
Can't we just leave this till the `mark` call? We're already performing the `originalPositionFor` call there. I think we can just store the name name we think, then prioritize the original's name during `mark`.